modules/renter/proto/filesection.go

```diff
@@ -0,0 +1,94 @@
+package proto
```

Comment:
This file needs tests to prove correctness. Especially around all of the edge cases that it's supposed to protect against.